packages/use-contractkit/src/connectors/connectors.ts

```diff
@@ -257,6 +259,7 @@ export class InjectedConnector implements Connector {
 
   async updateKitWithNetwork(network: Network): Promise<void> {
     localStorage.setItem(localStorageKeys.lastUsedNetwork, network.name);
+    this.network = network;
```

Comment:
if this isnt set then when init is called it will use the previous network still